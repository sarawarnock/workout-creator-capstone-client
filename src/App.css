.App {
  text-align: center;
  color: black;
}

.error {
  color: rgb(130, 0, 0);
}

/* HEADER AREA */
header {
  display: fixed;
  top: 0;
  width: 100%;
  padding: 40px;
}

header nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

h1 {
  font-family: Helvetica Neue;
  font-style: normal;
  font-weight: 500;
  font-size: 48px;
  line-height: 59px;
}

.home-link {
  width: 300px;
  text-align: left;
}

.home-link:hover {
  background-color:  #E5E5E5;
}

h2 {
  font-style: normal;
  font-weight: normal;
  font-size: 40px;
  line-height: 57px;
}

nav {
  display: grid;
  grid-area: nav;
  justify-self: center;
}

.header-links {
  height: 50px;
  width: 120px;
  border-radius: 30px;
  background-color: #C4C4C4;
  padding: 10px 20px;
  font-size: 24px;
  line-height: 29px;
}

.signup, .workouts {
  margin-right: 50px;
}

.workouts:hover {
  background: #006BFF;
  color: #fff;
}

.logout {
  background: #006BFF;
  color: #fff;
}

.logout:hover {
  background:#fff;
  color: #006BFF;
  border: solid 1px #006BFF;
}

/* .logout {
  background-color: #fff;
  color: #006BFF;
  border: solid 1.5px #006BFF;
}

.logout:hover {
  background-color: #006BFF;
  color: white;
  border: solid 1.5px #006BFF;
} */


/* MAIN PAGE */
main {
  margin: 0 auto;
  padding: 0 40px;
  max-width: 1080px;
}

.landing-text {
  margin: 25px auto 85px;
  width: 589px;
  text-align: left;
}

main p {
  font-family: Helvetica Neue;
  font-style: normal;
  font-weight: normal;
  font-size: 24px;
  line-height: 29px;
  margin-bottom: 25px;
}

.link {
  width: 200px;
  height: 50px;
  border-radius: 30px;
  background-color: #C4C4C4;
  padding: 10px 40px;
  font-size: 24px;
  line-height: 29px;
  margin-top: 40px;
}

/* LOGIN / SIGN UP */
.form-container {
  margin: 0 auto;
}

.inp-cont {
  margin: 0 auto;
}

.form-label {
  font-size: 24px;
  line-height: 29px;
  border: none;
}

.login-form input, .signup-form input {
  width: 403px;
  height: 80px;
  background:#E7F0FC;
  opacity: 0.9;
  font-size: 24px;
  line-height: 29px;
  padding: 0 10px;
  border: none;
  border-radius: 4px;
}

.login-form input:-webkit-autofill,  .signup-form input:-webkit-autofill {
  background-color: #C4C4C4;
  font-size: 24px;
}

.user-label {
  border: none;
}


/* CREATE WORKOUT */
.create-form {
  text-align: center;
}

.row {
  /* text-align: center; */
  margin: 0 auto;
  max-width: 500px;
}
/* 
#C4C4C4 
#5a98f0
#006BFF
*/

label {
  display: grid;
  align-self: center;
  margin: 10px;
  padding: 10px;
  font-size: 18px;
  font-weight: bolder;
  border: 1px solid rgb(14, 74, 159);
  color: rgb(14, 74, 159);
  border-radius: 4px;
}

.check {
  background-color: #fff;
}

input:checked + label {
  background-color: rgb(14, 74, 159);;
  font-weight: bolder;
  border: 1px solid rgb(14, 74, 159);
  color: #fff;
}

.checkbox-input {
  background-color: #fff;
  border: 1px solid rgb(14, 74, 159);
  color: rgb(14, 74, 159);
}

.selected-checkbox {
  background-color:rgb(14, 74, 159);
  font-weight: bolder;
  height: auto;
  margin: 10px;
  border: 1.5px solid rgb(14, 74, 159);
  color: #fff;
}

input[type="checkbox"] {
  display: none;
}

input[type="radio"] {
  display: none;
}

.name-wo {
  padding: 5px;
  font-size: 24px;
  border-radius: 4px;
  width: 30rem;
  margin  : 10px;
  border: solid 2px rgb(14, 74, 159);
}

.name-wo::placeholder {
  color: #C4C4C4;
}

.arr {
  color: #fff;
  border: solid 1.5px rgb(14, 74, 159);
  background-color: rgb(14, 74, 159);
}

.arr:hover{
  background: rgb(14, 74, 159);
  transform: scale(1.1);
  border: solid 1.5px rgb(14, 74, 159);
}

/* WORKOUTS LIST */
.workouts-header {
  margin-bottom: 20px;
}

.menu {
  text-align: center;
  padding: 0;
}

.row {
  display: grid;
  grid-area: main;
}

button, .view-link {
background-color: #C4C4C4;
border-radius:6px;
border:1px solid #C4C4C4;
display:inline-block;
cursor:pointer;
color:black;
font-family:Arial;
font-weight:bold;
padding:6px 24px;
text-decoration:none;
text-shadow:0px 1px 0px #C4C4C4;
}
a:hover, button:hover {
background-color: #a2a2a2;
transition: .3s;
}
a:active, button:active {
position:relative;
top:1px;
transition: .3s;
}

.btn {
  margin: 10px;
  font-size: 18px
}

.workouts-list {
  font-size: 24px;
}

.workouts-table {
  margin: 0 auto;
  table-layout: fixed;
  padding: 30px 0;
  min-width: 720px;
}

.tr {
  border-bottom: .5px solid black;
}

.tb-name, .tb-del  {
  width: 30%;
  font-size: 20px;
}

.tb-view {
  width: 30%;
  margin: 0 40px;
}

.wo-name {
  font-size: 22px;
}

.create-link {
  margin-top: 10px;
}

.create-link:hover {
  background: #006BFF;
  color: #fff;
}

/* VIEW WORKOUT */
.flex-cnt {
  display: flex;
  justify-content: space-between;
}

.tbh {
  vertical-align: top;
}

.inst {
  font-size: 1.8rem
}

.ic {
  display: flex;
  align-items: center;
}

.dt {
  font-size: 1.8rem;
}

.f-left {
  float: left;
}

.ml-10 {
  margin-left: 10px;
}

.lrg {
  font-size: 28px;
  font-weight: 600;
}

.left {
  text-align: left;
}

.right {
  text-align: right;
}

.up-c-left {
  text-transform: uppercase;
  text-align: left;
}

.num {
  font-size: 1.5rem;
}

.desc {
  font-size: 1.5rem;
}

.nav-link {
  list-style-type: none;
  color: white;
}

.navbar {
  color: white;
  align-self: start;
}

.nav {
  text-align: center;
  padding: 20px;
}

.detail {
  font-size: 20px;
}

.workout-details {
  font-size: 20px;
  padding: 10px;
  margin: 10px;
}

.exercise-reps {
  color: white;
  font-size: 24px;
}

.exercise-desc {
  font-size: 18px;
}

/* Exercise Screens */
.cap {
  text-transform: uppercase;
}

.ex-cont {
  margin: 20px auto;
  max-width: 560px;
  display: flex;
}

.ex-desc {
  margin: 0 10px 0 0;
  font-size: 24px;
}

.ex-reps {
  margin: 0 0 0 10px;
  font-size: 3rem;
}

/* STOPWATCH */
.sw-cont {
  margin: 40px;
}

.sw-cont h4{
  margin: 10px;
  font-size: 6rem;
}

h5 {
  margin: 40px;
  font-size: 2rem;
}

.btn-cont  button{
  margin: 0 5px;
}

/* @media screen and (max-width: 600px) {
  .App {
      display: grid;
      grid-template-columns: 15% 35% 35% 15%;
      grid-template-areas: 
      ". nav nav ."
      ". header header ."
      ". main main ."
  }

  header {
      grid-area: header;
  }

  nav {
      display: grid;
      grid-area: nav;
      justify-self: center;
  }

  main {
      display: grid;
      grid-area: main;
      padding: 10px;
  }

  .row {
      display: grid;
      grid-area: main;
  }

  .nav-link {
      list-style-type: none;
      color: white;
  }

  .nav label {
      display: inline-block;
      width: 100%; 
      color: white;
  }

  .nav ul li {
      display: block;
      cursor: pointer;
  }

  label {
      display: grid;
      align-self: center;
      border: 1px solid white;
      margin: 10px;
      padding: 10px;
      font-size: 18px;
  }
}  */

/* @media screen and (min-width: 600px) {
  .App {
      display: grid;
      grid-template-columns: 15% 35% 35% 15%;
      grid-template-areas: 
      "header header . ."
      "sidebar main main ."
  }

  .nav-link {
      list-style-type: none;
      color: white;
  }

  .menu {
      text-align: center;
  }
  
  .navbar {
      color: white;
      align-self: start;
  }

  main {
      display: grid;
      grid-area: main;
      padding: 10px;
  }

  .row {
      display: grid;
      grid-area: main;
  }

  label {
      display: grid;
      align-self: center;
  }
} */