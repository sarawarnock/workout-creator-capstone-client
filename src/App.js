import React from 'react';
import { Route, Link, Switch } from 'react-router-dom'
import config from './config'
import TokenService from './services/token-service'
import './App2.css';
import LandingPage from './landing-page'
import Navbar from './navbar'
import PersonalizedHomePage from './personalized-home-page'
import PastWorkouts from './past-workouts'
import Header from './header'
import Login from './log-in';
import SignUp from './sign-up';
import ViewPastWorkout from './view-past-workout';
import CreateNewWorkout from './create-new-workout'
import NotFoundPage from './not-found-page'

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      first_name: '',
      appSavedWorkouts: [],
      appSavedWorkoutDetails: []
    }
  }

  componentDidMount(){
    //get workouts by user ID
    //if the user has not logged in, get all past workouts, once user logs in, get their past workouts
    let getWorkoutUrl = ''
    if (sessionStorage.user_id == undefined) {
      getWorkoutUrl = `${config.API_ENDPOINT}/workouts`
    } else 
    getWorkoutUrl = `${config.API_ENDPOINT}/workouts/user/${TokenService.getUserId()}`;
    console.log(sessionStorage.user_id)
    fetch(getWorkoutUrl)
        .then(response => response.json())
        //map over the workouts by ID, returning each workout
        //so that we can get the individual workout details for that workout (including the exercises)
        .then(workouts => {
            this.setState({
                appSavedWorkouts: workouts
            });
      })
      //get all workout details for all workouts to use in Past Workouts component
      let getWorkoutDetailsUrl = `${config.API_ENDPOINT}/workoutdetails/workout/`;
        fetch(getWorkoutDetailsUrl)
        .then(response => response.json())
        .then(workoutDetails => {
            this.setState({
              appSavedWorkoutDetails: workoutDetails
            });
        })
        .catch(error => this.setState({ error }))
  }

  updateAppSavedWorkouts = newWorkout => {
    this.setState({
      appSavedWorkouts: [...this.state.appSavedWorkouts, newWorkout]
    })
  }
  
  //renders the Navbar
  renderNav() {
    return (
      <div>
        <Navbar />
      </div>
    )
  }

  //renders all of the routes 
  renderMainPages = () => {
    return (
      <div className="main-pages">
      <Switch>        
        <Route 
          exact
          path='/'
          component={LandingPage}
        />
        <Route 
          exact
          path={`/home`}
          component={PersonalizedHomePage}
        />
        <Route 
          exact
          path='/create-workout'
          render={(props) => <CreateNewWorkout {...props} saveNewWorkout={this.updateAppSavedWorkouts} />}
        />
        <Route 
          exact
          path={`/past-workouts`}
          render={(props) => <PastWorkouts {...props} appSavedWorkouts={this.state.appSavedWorkouts} 
            appSavedWorkoutDetails={this.state.appSavedWorkoutDetails}
          />}
        />
        <Route 
          exact
          path='/login'
          component={Login}
        />
        <Route 
          exact
          path='/sign-up'
          component={SignUp}
        />
        <Route 
          exact
          path='/past-workouts/:workout_id'
          render={(props) => <ViewPastWorkout {...props}
          appSavedWorkoutDetails={this.state.appSavedWorkoutDetails} />}
        />
        <Route 
          component={NotFoundPage}
        />
      </Switch>
      </div>
    )
  }

  render() {
    return (
      <div className="App">
        <nav className="app-nav"> {this.renderNav()} </nav>
        <header>
            <Link to='/'>
              <Header />
            </Link>
        </header>

        <main>
          {this.renderMainPages()}
        </main>
      </div>
    );
  }
}

export default App;

// helo