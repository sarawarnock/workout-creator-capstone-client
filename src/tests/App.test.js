import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { BrowserRouter } from 'react-router-dom'
import App from '../app'
import Login from '../log-in'
import SignUp from '../sign-up'
import NotFound from '../not-found-page'
import CreateNewWorkout from '../create-new-workout'
import PastWorkouts from '../past-workouts'
import ViewPastWorkout from '../view-past-workout'
import PersonalizedHomePage from '../personalized-home-page'
import Navbar from '../navbar'
import Header from '../header'
import LandingPage from '../landing-page'

const testWorkoutsArray = [
  {
    id: 1,
    total_length: 10,
    user_id: 1,
    workout_type: 'EMOM',
    workouts_name: 'Test workout'
  },
  {
    id: 2,
    total_length: 15,
    user_id: 1,
    workout_type: 'AMRAP',
    workouts_name: 'Test workout 2'
  }
]

const testWorkoutDetailsArray = [
  {
    description: 'Test description',
    exercise_reps: 10,
    exercises_id: 1,
    title: 'test title',
    workouts_id: 1
  },
  {
    description: 'Test description',
    exercise_reps: 10,
    exercises_id: 2,
    title: 'test title',
    workouts_id: 1
  }
]

const params =  () => {}
const goBack = () => {}

describe('Workout Creator app', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <App />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe('New Workout component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CreateNewWorkout />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<CreateNewWorkout />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})

describe('Past Workouts componenet', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter history={goBack}>
        <PastWorkouts appSavedWorkouts={testWorkoutsArray} match= { params }  />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter history={goBack}>
          <PastWorkouts appSavedWorkouts={testWorkoutsArray} match={ params } />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})

describe('View Past Workouts componenet', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter >
        <ViewPastWorkout appSavedWorkoutDetails={testWorkoutDetailsArray} history={goBack} match={{ params }} />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <ViewPastWorkout appSavedWorkoutDetails={testWorkoutDetailsArray} history={goBack} match={{ params }} />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Login component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <Login />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <Login />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Sign Up component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <SignUp />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <SignUp />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Not Found component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <NotFound />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Header component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Navbar component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Personal Home Page component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <PersonalizedHomePage />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <PersonalizedHomePage />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Landing Page component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <LandingPage />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <LandingPage />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});