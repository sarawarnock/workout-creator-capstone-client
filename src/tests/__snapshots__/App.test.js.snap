// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the UI as expected 1`] = `
<div
  className="App"
>
  <main
    className="main"
  >
    <h1>
      MyMetcon Workout Creator
    </h1>
  </main>
</div>
`;

exports[`Landing Page component renders the UI as expected 1`] = `
<div
  className="App"
>
  <main
    className="main"
  >
    <h1>
      Workout Creator
    </h1>
    <p>
      Create your own metabolic-conditoning workout
    </p>
    <p>
      Choose how long you want to workout for and the muscle groups you want to work, and we'll do the rest!
    </p>
    <button
      className="big-btn"
    >
       
      <a
        href="/sign-up"
        onClick={[Function]}
      >
         Sign Up 
      </a>
    </button>
    <button
      className="big-btn"
    >
       
      <a
        href="/login"
        onClick={[Function]}
      >
         Login
      </a>
       
    </button>
    <p>
      To login: 
    </p>
    <p>
      email: testuser@gmail.com
    </p>
    <p>
      password: Testpassword1
    </p>
  </main>
</div>
`;

exports[`Login component renders the UI as expected 1`] = `
<div
  className="App"
>
  <main>
    <h1>
      Log In
    </h1>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <label
        className="user-label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        name="loginEmail"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="text"
      />
      <label
        className="user-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        name="loginPassword"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
      <button
        className="small-btn"
        type="submit"
      >
         Log In 
      </button>
    </form>
    <div>
      <h2>
        Don't have an account yet?
      </h2>
      <a
        href="/sign-up"
        onClick={[Function]}
      >
        <button
          className="small-btn"
        >
          Sign Up
        </button>
      </a>
    </div>
  </main>
</div>
`;

exports[`Navbar component renders the UI as expected 1`] = `
<div
  className="navbar"
>
  <nav
    className="nav"
    role="navigation"
  >
    
  </nav>
</div>
`;

exports[`New Workout component renders the UI as expected 1`] = `
<div
  className="App"
>
  <div
    className="row"
  >
    <div
      className="col-sm-12"
    >
      <form
        onSubmit={[Function]}
      >
        <h2
          className="workout-question-1"
        >
          Which muscle groups would you like to work?
        </h2>
        <div
          className="checkbox"
        >
          <div
            className="App"
          >
            <div
              className="row"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  className="form-check-input"
                  name="Arms"
                  onChange={[Function]}
                  type="checkbox"
                />
                Arms
              </label>
            </div>
          </div>
          <div
            className="App"
          >
            <div
              className="row"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  className="form-check-input"
                  name="Legs"
                  onChange={[Function]}
                  type="checkbox"
                />
                Legs
              </label>
            </div>
          </div>
          <div
            className="App"
          >
            <div
              className="row"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  className="form-check-input"
                  name="Chest"
                  onChange={[Function]}
                  type="checkbox"
                />
                Chest
              </label>
            </div>
          </div>
          <div
            className="App"
          >
            <div
              className="row"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  className="form-check-input"
                  name="Back"
                  onChange={[Function]}
                  type="checkbox"
                />
                Back
              </label>
            </div>
          </div>
          <div
            className="App"
          >
            <div
              className="row"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  className="form-check-input"
                  name="Core"
                  onChange={[Function]}
                  type="checkbox"
                />
                Core
              </label>
            </div>
          </div>
          <div
            className="App"
          >
            <div
              className="row"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  className="form-check-input"
                  name="Cardio"
                  onChange={[Function]}
                  type="checkbox"
                />
                Cardio
              </label>
            </div>
          </div>
          <div
            className="App"
          >
            <div
              className="row"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  className="form-check-input"
                  name="Advanced"
                  onChange={[Function]}
                  type="checkbox"
                />
                Advanced
              </label>
            </div>
          </div>
        </div>
        <h2
          className="workout-question-2"
        >
          How long do you want to workout for?
        </h2>
        <div
          className="workout-time answer-2"
        >
          <input
            id="time-1"
            name="workoutTimeValue"
            onClick={[Function]}
            type="radio"
            value="5"
          />
          <label
            htmlFor="time-1"
          >
            5 minutes
          </label>
          <input
            id="time-2"
            name="workoutTimeValue"
            type="radio"
            value="10"
          />
          <label
            htmlFor="time-2"
          >
            10 minutes
          </label>
          <input
            id="time-3"
            name="workoutTimeValue"
            type="radio"
            value="15"
          />
          <label
            htmlFor="time-3"
          >
            15 minutes
          </label>
          <input
            id="time-4"
            name="workoutTimeValue"
            type="radio"
            value="20"
          />
          <label
            htmlFor="time-4"
          >
            20 minutes
          </label>
          <input
            id="time-5"
            name="workoutTimeValue"
            type="radio"
            value="25"
          />
          <label
            htmlFor="time-5"
          >
            25 minutes
          </label>
          <input
            id="time-6"
            name="workoutTimeValue"
            type="radio"
            value="30"
          />
          <label
            htmlFor="time-6"
          >
            30 minutes
          </label>
        </div>
        <h2
          className="workout-question-3"
        >
          What style of workout would you like?
        </h2>
        <div
          className="workout-type answer-3"
        >
          <input
            id="emom"
            name="workoutTypeValue"
            onClick={[Function]}
            type="radio"
            value="EMOM"
          />
          <label
            htmlFor="emom"
          >
            EMOM (Every Minute On the Minute)
          </label>
          <input
            id="amrap"
            name="workoutTypeValue"
            type="radio"
            value="AMRAP"
          />
          <label
            htmlFor="amrap"
          >
            AMRAP (As Many Rounds As Possible)
          </label>
        </div>
        <br />
        <h2>
          Name Your Workout:
        </h2>
        <div
          className="workouts-name"
        >
          <input
            id="workouts-name-input"
            name="workoutNameValue"
            onChange={[Function]}
            placeholder="Example: Workout 1"
            type="text"
          />
        </div>
        <br />
        <button
          className="big-btn"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Not Found component renders the UI as expected 1`] = `
<div>
  <h1>
    Page Not Found!
  </h1>
</div>
`;

exports[`Past Workouts componenet renders the UI as expected 1`] = `
<div
  className="App"
>
  <main
    className="main"
  >
    <h1>
      Past Workouts
    </h1>
    <table
      className="workouts-table"
    >
      <thead>
        <tr>
          <th
            className="tb-name"
          >
            Name
          </th>
          <th
            className="tb-id"
          >
            Workout ID
          </th>
          <th
            className="tb-view"
          >
            View Workout
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Test workout
          </td>
          <td>
             
            1
             
          </td>
          <td>
            <button>
              <a
                href="/past-workouts/1"
                onClick={[Function]}
              >
                View
              </a>
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Test workout 2
          </td>
          <td>
             
            2
             
          </td>
          <td>
            <button>
              <a
                href="/past-workouts/2"
                onClick={[Function]}
              >
                View
              </a>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <a
      href="/create-workout"
      onClick={[Function]}
    >
      <button>
        Create Workout
      </button>
    </a>
  </main>
</div>
`;

exports[`Personal Home Page component renders the UI as expected 1`] = `
<div
  className="App"
>
  <main>
    <h1>
      Home
    </h1>
    <div
      className="workout-btn"
    >
      <button
        className="big-btn"
      >
        <a
          href="/past-workouts"
          onClick={[Function]}
        >
          Past Workouts
        </a>
      </button>
      <button
        className="big-btn"
      >
        <a
          href="/create-workout"
          onClick={[Function]}
        >
          New Workout
        </a>
      </button>
    </div>
  </main>
</div>
`;

exports[`Sign Up component renders the UI as expected 1`] = `
<div
  className="App"
>
  <main>
    <h1>
      Sign Up
    </h1>
    <form
      className="sign-up-form"
      onSubmit={[Function]}
    >
      <div
        className="sign-up-email"
      >
        <label
          className="user-label"
          htmlFor="email"
        >
          Email
        </label>
        <input
          id="email"
          name="signUpEmail"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="text"
        />
      </div>
      <div
        className="sign-up-password"
      >
        <label
          className="user-label"
          htmlFor="password"
        >
          Password (one capital letter and one number needed)
        </label>
        <input
          id="password"
          name="signUpPassword"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="sign-up-name"
      >
        <label
          className="user-label"
          htmlFor="fname"
        >
          First Name
        </label>
        <input
          id="fname"
          name="signUpFirstName"
          onChange={[Function]}
          placeholder="First Name"
          required={true}
          type="text"
        />
      </div>
      <button
        className="small-btn"
        type="submit"
      >
        Sign Up!
      </button>
    </form>
    <div>
      <h2>
        Already have an account?
      </h2>
      <button
        className="small-btn"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </button>
    </div>
  </main>
</div>
`;

exports[`View Past Workouts componenet renders the UI as expected 1`] = `
<div
  className="App"
>
  <main
    className="main"
  >
    <a
      className="NavCircleButton "
    >
      Go Back
    </a>
  </main>
</div>
`;
